applications/tari_base_node/src/bootstrap.rs

```diff
@@ -106,6 +106,10 @@ where B: BlockchainBackend + 'static
             .map_err(|e| ExitError::new(ExitCode::ConfigError, e))?;
         p2p_config.transport.tor.identity = tor_identity;
 
+        // TODO: This should probably be disabled in future and have it optionally set/unset in the config - this check
```

Comment:
Can you add more detail about how this leaks the IP address?

Reply:
Sure, TL;DR is that the "pingpong" text is periodically sent on an unencrypted socket allowing anyone observing the traffic to recognise the sending IP address as a tari node. With tor, a tor router can make this observation and over IP an ISP/MITM. In fairness, this observation can also be made by observing the first network byte.